lib/src/widgets/yaru_choice_chip_bar.dart

```diff
@@ -86,63 +86,140 @@ class _YaruChoiceChipBarState extends State<YaruChoiceChipBar> {
           )
     ];
 
-    if (widget.wrap) {
+    final listView = ListView(
+      scrollDirection: Axis.horizontal,
+      controller: _controller,
+      children: children
+          .map(
+            (e) => Padding(
+              padding: EdgeInsets.only(
+                right: widget.spacing,
+              ),
+              child: e,
+            ),
+          )
+          .toList(),
+    );
+
+    final borderRadius = BorderRadius.circular(8);
+    final roundedRectangleBorder = RoundedRectangleBorder(
+      borderRadius: borderRadius,
+      side: BorderSide(
+        color: theme.colorScheme.outline,
+        width: 1,
+      ),
+    );
+    const clipRadius = 40.0;
+    const navigationButtonSize = 34.0;
+
+    final goPreviousButton = SizedBox(
+      height: navigationButtonSize,
+      width: navigationButtonSize,
+      child: Material(
+        shape: roundedRectangleBorder,
+        child: InkWell(
+          customBorder: roundedRectangleBorder,
+          onTap: _enableGoPreviousButton
+              ? () => _controller.animateTo(
+                    _controller.position.pixels - widget.animationStep,
+                    duration: widget.duration,
+                    curve: Curves.bounceIn,
+                  )
+              : null,
+          child: const Icon(YaruIcons.go_previous),
+        ),
+      ),
+    );
+
+    final goNextButton = SizedBox(
+      height: navigationButtonSize,
+      width: navigationButtonSize,
+      child: Material(
+        shape: roundedRectangleBorder,
+        child: InkWell(
+          customBorder: roundedRectangleBorder,
+          onTap: _enableGoNextButton
+              ? () => _controller.animateTo(
+                    _controller.position.pixels + widget.animationStep,
+                    duration: widget.duration,
+                    curve: Curves.bounceIn,
+                  )
+              : null,
+          child: const Icon(YaruIcons.go_next),
+        ),
+      ),
+    );
+
```

Comment:
Those two buttons are basically the same except the icon and the animation position.
Maybe you can create a factory function at the same place in `build()` with the icon and the animation step as parameter?

Reply:
I extracted the buttons but I think keeping the animated opacit in place of the build makes it more readble. Ok for you?

Reply:
I agree :+1: